twake/backend/node/src/services/files/services/preview.ts

```diff
@@ -0,0 +1,73 @@
+import { FilePubsubHandler, FileServiceAPI } from "../api";
+import { logger, TwakeContext } from "../../../core/platform/framework";
+import _ from "lodash";
+import { PubsubServiceAPI } from "../../../core/platform/services/pubsub/api";
+import { PreviewPubsubCallback } from "../../../../src/services/previews/types";
+import Repository from "../../../../src/core/platform/services/database/services/orm/repository/repository";
+import { File } from "../entities/file";
+
+/**
+ * Update the file metadata and upload the thumbnails in storage
+ */
+export class PreviewFinishedProcessor implements FilePubsubHandler<PreviewPubsubCallback, string> {
+  repository: Repository<File>;
+
+  constructor(
+    readonly service: FileServiceAPI,
+    private pubsub: PubsubServiceAPI,
+    repository: Repository<File>,
+  ) {
+    this.service = service;
+    this.repository = repository;
+  }
```

Comment:
All of this can be simplified by using parameters properties: https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties